CMS/cms-basics

```diff
@@ -0,0 +1,345 @@
+```
+
+## Example: configuring a basic CMS
+
+In the following examples, we will be using the below database structure as a sample.
+
+|Record	|Fields	|	|	|	|	|
+|---	|---	|---	|---	|---	|---
+|user	|_id	|_created_at	|username	|role	|avatar	|
+|blogpost	|_id	|_created_at	|title	|content	|
+|comment	|_id	|_created_at	|comment	|blogpostId (Skygear reference)	|
+
+### Site
+
+**Site** defines what pages to be shown in the CMS. All availabe pages will be shown on the left sidebar.
+
+There are 3 types of pages you can show:
+
+1. Record: a page to display all the records of a record type.
+2. Push Notifications: a page to broadcast push messages to your users. (Coming soon)
+3. User management: a page to manage your users, from creating a new user account, to blocking problemtics users. (Coming soon)
+
+For example, if you want to display all the records from **user** and the **blogpost**, you should write in the YML:
+
+```
+site:
+  - type: Record
+    name: user
+    label: User
+  - type: Record
+    name: blogpost
+    label: Blogpost
+```
+
+`name` refers to the record type in the database. `label` refers to the name to be shown on the CMS.
+
+If you only have 'Site' in the YML config, you will only see a blank page in the CMS because we haven't defined what to display on the pages. Let see how to do it below  👇
+
+
+### Records
+
+By now we have already defined two pages - **User** & **Blogpost**. Next we should define what to show in these pages.
+
+In general there are **FOUR** views you can customize: **list, show, edit and new**.
+
+* **List**: displayable fields on the first page. you can further set which fields can be **filtered**
+* **Show**: dispalyable fields on the detail page
+* **Edit**: editable fields to be editable
+* **New**: fill-in-able fields when creating a new record
+
+#### List
+
+Let's first define the fields we want to show in the `user` page. These are the two goals:
+
+1. I want to display only the **username**, the  **role**, the **avatar** and the **_created_at** field of a `user` record.
+2. I also want **role** to be searchable / filterable.
+
+```
+site:
+  - type: Record
+    name: user
+    label: User
+  - type: Record
+    name: blogpost
+    label: Blogpost
+
+records:
+  user:
+    list:
+      fields:
+        - name: username
+          type: String
+          label: Username
+        - name: role
+          type: Dropdown
+          label: Email
+        - name: avatar
+          type: ImageAsset
+          label: Avatar
+        - type: _created_at
+      filters:
+        - name: username
+          type: String
+          label: Username
+
+```
+
+Now in your CMS, you should be able to see all your users records!
+
+Again, `name` is the field name. `label` is the name to be shown on the CMS.
+
+However, if you click on the button `Show`, you will get a 'Page Not Found' Error. This will be fixed when we configure the `Show` view of the user record.
```

Comment:
This is probably a bug, and not explained correctly in guide.
1. if `show:` section was not defined, it should not show button, so you can't click it and got "Page Not Found" error.
1. Originally when I talk with @limouren , I remember (I hope it wasn't missing in details...) the conclusion was to help make writing the config easier, if a section is defined as empty, like below, the default behavior is to show all fields with default type.

e.g.

```
records:
  user:
     list: {} # omit details
     show: ~
```

cc: @Steven-Chan @carmenlau